packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -355,6 +361,55 @@ def __eq__(self, other: Any) -> bool:
         return self.payload == other.payload and self.signature == other.signature
 
 
+@dataclass(frozen=True)
+class NewBaseTxPayload(ABC, metaclass=_MetaPayload):
+    """This class represents a base class for transaction payload classes."""
+
+    sender: str
+    round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
+
+
+@dataclass
+class NewTransaction:
+    """Class to represent a transaction for the ephemeral chain of a period."""
+
+    payload: NewBaseTxPayload
+    signature: Optional[str] = None
+
+    def __bytes__(self) -> bytes:
+        """Serialize the transaction to bytes"""
+
+        payload = {self.payload.__class__.__name__: asdict(self.payload)}
+        tx_data = dict(payload=payload, signature=self.signature)
+        encoded_data = json.dumps(tx_data, sort_keys=True).encode()
+        if sys.getsizeof(encoded_data) > MAX_READ_IN_BYTES:
+            msg = f"Transaction must be smaller than {MAX_READ_IN_BYTES} bytes"
+            raise ValueError(msg)
+        return encoded_data
```

Comment:
simply serialize to json. Note that, unlike with pickle, this does not allow for a one-to-one mapping between datastructures (e.g. `assert isinstance(json.loads(json.dumps(tuple())), list)`).